examples/hosts/app-integration/external-data/src/view/debugView.tsx

```diff
@@ -57,35 +60,49 @@ export interface IDebugViewProps {
  * For the purposes of this test app, it is useful to be able to see both data sources side-by-side.
  */
 export const DebugView: React.FC<IDebugViewProps> = (props: IDebugViewProps) => {
+	const { model, unresolvedChanges } = props;
+	// Flag that represents the state in which clients are actively fetching external data.
+	const [fetchingExternalData, setFetchingExternalData] = useState(false);
 	return (
 		<div>
 			<h2 style={{ textDecoration: "underline" }}>External Data Server App</h2>
-			<TaskListView model={props.model} />
-			<ExternalDataView />
-			<SyncStatusView />
-			<ControlsView model={props.model} />
+			<ExternalAppForm model={model} setFetchingExternalData={setFetchingExternalData} />
+			<ExternalDataView setFetchingExternalData={setFetchingExternalData} />
+			<SyncStatusView
+				fetchingData={fetchingExternalData}
+				unresolvedChanges={unresolvedChanges}
+			/>
+			<ControlsView model={model} />
 		</div>
 	);
 };
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-interface IExternalDataViewProps {}
+interface IExternalDataViewProps {
+	setFetchingExternalData: React.Dispatch<React.SetStateAction<boolean>>;
+}
 
 const ExternalDataView: React.FC<IExternalDataViewProps> = (props: IExternalDataViewProps) => {
+	const { setFetchingExternalData } = props;
 	const [externalData, setExternalData] = useState({});
 	useEffect(() => {
 		// Run once immediately to run without waiting.
-		pollForServiceUpdates(externalData, setExternalData).catch(console.error);
+		pollForServiceUpdates(externalData, setExternalData, setFetchingExternalData).catch(
+			console.error,
+		);
 
 		// HACK: Poll every 3 seconds
 		const timer = setInterval(() => {
-			pollForServiceUpdates(externalData, setExternalData).catch(console.error);
+			setFetchingExternalData(true);
```

Comment:
I don't think you want these calls to `setFetchingExternalData` here, since `pollForServiceUpdates` already does it.

Additionally, your second call where you set it to false will currently run *before* the promise returned from `pollForServiceUpdates` finishes running. This means that you're likely setting the boolean value back to false immediately after you set it to true.